lib/ex_aws/s3.ex

```diff
@@ -1006,15 +1006,16 @@ defmodule ExAws.S3 do
     {hash_header(alg), :crypto.hash(alg, content) |> Base.encode64()}
   end
 
-  @spec get_hash_config() :: :md5
+  @spec get_hash_config() :: atom()
```

Comment:
This could even be a fixed list of atoms if we know a bounded set of things this could be. If it's like 5 then you could be explicit, but if it's hundreds or dynamic or something else +1 to atom

Reply:
Ended up making this a type that only includes the algorithms S3 supports